test/2_test_fullpath_with_permit.ts

```diff
@@ -5252,6 +5252,205 @@ describe('Cashier and VoucherKernel', () => {
+          )
+            .to.emit(contractVoucherKernel, eventNames.LOG_VOUCHER_FAULT_CANCEL)
+            .withArgs(tokenVoucherIdCommited);
+          expect(
+            await bosonRouterOldSeller.cancelOrFault(tokenVoucherIdRedeemed)
+          )
+            .to.emit(contractVoucherKernel, eventNames.LOG_VOUCHER_FAULT_CANCEL)
+            .withArgs(tokenVoucherIdRedeemed);
+          expect(
+            await bosonRouterOldSeller.cancelOrFault(tokenVoucherIdRefunded)
+          )
+            .to.emit(contractVoucherKernel, eventNames.LOG_VOUCHER_FAULT_CANCEL)
+            .withArgs(tokenVoucherIdRefunded);
+        });
+
+        it('After transfer, old owner gets correct amounts when voucher finalised', async () => {
+          await advanceTimeSeconds(2 * constants.SECONDS_IN_DAY);
+          await utils.expire(tokenVoucherIdCommited, users.deployer.signer);
+          await advanceTimeSeconds(60);
+          await utils.finalize(tokenVoucherIdCommited, users.deployer.signer);
+          await utils.finalize(tokenVoucherIdRedeemed, users.deployer.signer);
+          await utils.finalize(tokenVoucherIdRefunded, users.deployer.signer);
+
+          const oldOwnerFromEscrow = await contractCashier.getEscrowAmount(
+            users.other1.address
+          );
+
+          const oldOwnerBalance = await ethers.provider.getBalance(
+            users.other1.address
+          );
+
+          await utils.withdraw(tokenVoucherIdCommited, users.deployer.signer);
+          let expectedBalance = oldOwnerBalance.add(
+            constants.PROMISE_DEPOSITSE1
+          );
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(constants.PROMISE_DEPOSITSE1),
+            'Wrong escrow amount after expiry'
+          );
+          expect(
+            await ethers.provider.getBalance(users.other1.address)
+          ).to.equal(expectedBalance, 'Wrong withdrawn amount after expiry');
+
+          await utils.withdraw(tokenVoucherIdRedeemed, users.deployer.signer);
+          expectedBalance = expectedBalance.add(
+            constants.PROMISE_DEPOSITSE1 + constants.PROMISE_PRICE1
+          );
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(2 * constants.PROMISE_DEPOSITSE1),
+            'Wrong escrow amount after redeem'
+          );
+          expect(
+            await ethers.provider.getBalance(users.other1.address)
+          ).to.equal(expectedBalance, 'Wrong withdrawn amount after redeem');
+
+          await utils.withdraw(tokenVoucherIdRefunded, users.deployer.signer);
+          expectedBalance = expectedBalance.add(constants.PROMISE_DEPOSITSE1);
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(3 * constants.PROMISE_DEPOSITSE1),
```

Comment:
Why is this `3 * constants.PROMISE_DEPOSITSE1` ?

Is it because `oldOwnerFromEscrow` is defined with `const` and the value never actually gets updated in the previous lines where `oldOwnerFromEscrow.sub` is called?

Reply:
I think that my answer for comment above explains this as well.

And yes, when calling `oldOwnerFromEscrow.sub` the value of `oldOwnerFromEscrow` actually stays intact/